pyccel/codegen/printing/fcode.py

```diff
@@ -2697,12 +2790,40 @@ def split_pos_code(line, endpos):
         result = []
         trailing = ' &'
         for line in lines:
-            if len(line)>72 and ('"' in line[72:] or "'" in line[72:] or '!' in line[:72]):
+            if len(line)>72 and '!' in line[:72]:
                 result.append(line)
-
+            elif len(line)>72 and ('"' in line[72:] or "'" in line[72:]):
+                pos=[]
+                quote = 0
+                double_quote = 0
+                mine = []
+                for idx, c in enumerate(line):
+                    if c == '\'':
+                        quote += 1
+                    if c == '\"':
+                        double_quote += 1
+                    if c == ' ' and quote%2 == 0 and double_quote %2 == 0 and idx > 72 and (not pos or idx > pos[-1] + 65):
+                        pos.append(idx)
+                last_cut = 0
+                for cut in pos:
+                    hunk = line[:(cut-last_cut)].rstrip()
+                    line = line[(cut-last_cut):].lstrip()
+                    if line:
+                        hunk += trailing
+                    if cut == pos[0]:
+                        result.append(hunk)
+                        mine.append(hunk)
+                    else:
+                        result.append("%s%s"%("      " , hunk))
+                        mine.append("%s%s"%("      " , hunk))
```

Comment:
I think this could be written more clearly as:
```
result.append("      " + hunk)
```
I don't see what advantage string formatting is giving you here

Reply:
that was the initial syntax. I ll rectify it.